domain/sysvar_cache.go

```diff
@@ -112,16 +108,22 @@ func (svc *SysVarCache) fetchTableValues(ctx sessionctx.Context) (map[string]str
 	return tableContents, nil
 }
 
-// RebuildSysVarCache rebuilds the sysvar cache both globally and for session vars.
+// rebuildSysVarCache rebuilds the sysvar cache both globally and for session vars.
 // It needs to be called when sysvars are added or removed.
-func (svc *SysVarCache) RebuildSysVarCache(ctx sessionctx.Context) error {
+func (do *Domain) rebuildSysVarCache() error {
 	newSessionCache := make(map[string]string)
 	newGlobalCache := make(map[string]string)
+	sysSessionPool := do.SysSessionPool()
+	ctx, err := sysSessionPool.Get()
```

Comment:
`GetGlobalSysVar` has had performance issues in the past, this is because in TiDB 5.1 and below it would read from the tikv table. So for example `SHOW VARIABLES` would call this hundreds of times in serial.

But, this code change should not impact `GetGlobalSysVar` except on server first start, when it is detected that the sysvar cache is empty (see `rebuildSysVarCacheIfNeeded()` logic).

But for your second point, it will affect `replaceGlobalVariablesTableValue`, since on the initiating server it will require a ctx to be `Get()`'ed.  I don't think this is a common code path, and while there is syntax to set multiple values at once:

```sql
mysql> set global autocommit =1, tidb_enable_noop_functions=1, sql_mode="";
Query OK, 0 rows affected (0.00 sec)
```

This is really uncommon to execute. It is typically not a hot code-path to update global variables (and there will be a tikv request at the same time to update the table, which will typically take longer.)